packages/helper-grammar-regex-collection/index.js

```diff
@@ -1,15 +1,81 @@
 import go from './go.js';
+import regexBuilder from './regexBuilder';
 
-const REQUIRE = /require(?:\.resolve)?(?:\s|\()\s*['"]([^'"\s]+)['"]\s*\)?/g;
-const IMPORT = /import [\r\n\s\w{},*\$]*(?: from )?['"]([^'"\s]+)['"]/g;
-const EXPORT = /export [\r\n\s\w{},*\$]*(?: from )['"]([^'"\s]+)['"]/g;
-const GEM = /gem ['"]([^'"\s]+)['"]/g;
-const HOMEBREW = /(?:depends_on|conflicts_with)(?: cask:| formula:)? ['"]([^'"\s]+)['"]/g;
-const TYPESCRIPT_REFERENCE = /\/{3}\s?<reference path=['"]([^'"\s]+)['"]/g;
-const DOCKER_FROM = /FROM\s([^\n]*)/g;
-const VIM_PLUGIN = /(?:(?:(?:Neo)?Bundle(?:Lazy|Fetch)?)|Plug(?:in)?)\s['"]([^'"\s]+)['"]/g;
-const RUST_CRATE = /(?:extern crate|use) ([^:; ]+)/g;
-const PYTHON_IMPORT = /^\s*(?:import|from)\s([^\s]*)/gm;
+const subpatterns = {};
+const regex = regexBuilder(subpatterns);
+
+subpatterns.captureQuotedWord = regex`
```

Comment:
I just pushed up a [refactor](https://github.com/OctoLinker/browser-extension/pull/282/files/54d73ecb030eb4a3a3f29a5a4ccf331f4ecd0caa..2c2d663e56642b35ec3d735d2ce7e241bbfcddc1) I had been working on, that addresses this by allowing normal template literal interpolation to be used, so that `subpatterns` isn't exposed outside of the `regex` tag function.

For what it's worth, this refactor was also done with the idea that I might eventually contribute it back to XRegExp. See https://github.com/slevithan/xregexp/issues/103